makeBinaryWrapper: add -Wno-overlength-strings

ncfavier opened the pull request:
The generated C code contains large string literals that are longer than the maximum length specified by the standard.

However, GCC has no trouble dealing with those strings, so we can just add -Wno-overlength-strings.

https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Woverlength-strings

Fixes errors like https://hydra.nixos.org/build/175508354/nixlog/1 caused by https://github.com/NixOS/nixpkgs/pull/164163.

<details>
<summary>For information, here's the generated C program for the <code>brave</code> wrapper</summary>

```c
#define _GNU_SOURCE         /* See feature_test_macros(7) */
#include <unistd.h>
#include <stdlib.h>
#include <assert.h>
#include <stdio.h>
#define assert_success(e) do { if ((e) < 0) { perror(#e); abort(); } } while (0)
void set_env_prefix(char *env, char *sep, char *prefix) {
    char *existing = getenv(env);
    if (existing) {
        char *val;
        assert_success(asprintf(&val, "%s%s%s", prefix, sep, existing));
        assert_success(setenv(env, val, 1));
        free(val);
    } else {
        assert_success(setenv(env, prefix, 1));
    }
}
int main(int argc, char **argv) {
    set_env_prefix("GIO_EXTRA_MODULES", ":", "/nix/store/q1yars9wd0qyx1ib78l4y6jc3cfdd5l8-dconf-0.40.0-lib/lib/gio/modules");
    set_env_prefix("GIO_EXTRA_MODULES", ":", "/nix/store/q1yars9wd0qyx1ib78l4y6jc3cfdd5l8-dconf-0.40.0-lib/lib/gio/modules");
    putenv("GDK_PIXBUF_MODULE_FILE=/nix/store/k6a58j0i1qy9vc87lppz1gwq29drr62b-librsvg-2.54.0/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache");
    set_env_prefix("XDG_DATA_DIRS", ":", "/nix/store/3a3j7xarxmrarxasi2sr05afx3vnx6mr-gtk+3-3.24.33/share/gsettings-schemas/gtk+3-3.24.33:/nix/store/snpvaw1f86407x7aw114khhpvmyjcr0g-gsettings-desktop-schemas-42.0/share/gsettings-schemas/gsettings-desktop-schemas-42.0");
    set_env_prefix("XDG_DATA_DIRS", ":", "/nix/store/vmyri2fwck810hsfm7cf4jaa2hv8mnzj-brave-1.38.109/share");
    set_env_prefix("LD_LIBRARY_PATH", ":", "/nix/store/a3nqf29wdxf2022ri3m3y8302mvlpfy7-alsa-lib-1.2.6.1/lib:/nix/store/60wrl1spfjm04hsqym5jqanc0sgz3qs3-at-spi2-atk-2.38.0/lib:/nix/store/5s9n8syy1fa22jxmg865xpivb63kp1hy-at-spi2-core-2.44.1/lib:/nix/store/v8j970izrm2agqqh19c6j23d25asip3d-atk-2.36.0/lib:/nix/store/fnad9qhmlzl9bgm2wh6mgv0fmrg33k32-cairo-1.16.0/lib:/nix/store/byzr6wj4dsh6y146jxxwrhc64lq92bmx-cups-2.4.1-lib/lib:/nix/store/hyqhzp899sr3zh0z1fsy8wl59mj2ib1l-dbus-1.12.20-lib/lib:/nix/store/yabw7v3z07l2n7jnhma0mjkr7h586a7b-expat-2.4.7/lib:/nix/store/4bhkj5w5319pig0n47l2qhalx2bz9f0c-fontconfig-2.13.94-lib/lib:/nix/store/pjfijj9j1pwcb76yqny64k8wk7yz191p-freetype-2.12.0/lib:/nix/store/nyxva3vqcx7331i61760h4sdkmxnrkzh-gdk-pixbuf-2.42.8/lib:/nix/store/hqvf17nn5fmwrlrgq9qyplz1id923jri-glib-2.72.1/lib:/nix/store/3a3j7xarxmrarxasi2sr05afx3vnx6mr-gtk+3-3.24.33/lib:/nix/store/5fl91pknshzhgv1147j2bfyr2jr77skq-libdrm-2.4.110/lib:/nix/store/3lrppnbkc5wlnh5ifwlxhghb0lxan42h-libX11-1.7.2/lib:/nix/store/xsi5z55nxm3n3m0pzysq8c6ccrqqhj4z-libGL-1.4.0/lib:/nix/store/8g1dgdpjgldsa4faj7rfnw06n04idhyz-libxkbcommon-1.4.0/lib:/nix/store/gwd0iijizj8k4ba5wpxnwjjgn1sfc1sp-libXScrnSaver-1.2.3/lib:/nix/store/zgjj5alj0qyhmgy45rfdbd35zgqw1syz-libXcomposite-0.4.5/lib:/nix/store/ipgwk0ywkbrwsfqi6s0nmbybqds84zkd-libXcursor-1.2.0/lib:/nix/store/fxah0qfxr07y3j2sldq2izfq3h5j1wiy-libXdamage-1.1.5/lib:/nix/store/91x9nnlbqpqg8mcb9qjlb1q9jkq9v7v3-libXext-1.3.4/lib:/nix/store/llnvb4l7krvq2s0rsbxb1r2xn3s8hz23-libXfixes-6.0.0/lib:/nix/store/nw60c3l2qq3akr1ld6z0rlzb9gw293h8-libXi-1.8/lib:/nix/store/85c71b4h8afbhp648mbm2lnq73ipmsc2-libXrandr-1.5.2/lib:/nix/store/89yf477csc3hyn7kpvc1nnj1731n2sml-libXrender-0.9.10/lib:/nix/store/h05nwiqmiz17484qcj30m4xjsm8yf5lx-libxshmfence-1.3/lib:/nix/store/2wa7lfl6kh0q9p3njxdjzlfkvv8lhmn0-libXtst-1.2.3/lib:/nix/store/ip19s520xvvxhpm69g355vbhgyvv4z7q-util-linux-minimal-2.37.4-lib/lib:/nix/store/cyxpd4cf7wbb5q18bcqkkyhriyajas11-mesa-22.0.2/lib:/nix/store/l6np4r402hwcj0kcqgrv5mqc819qnbah-nspr-4.33/lib:/nix/store/jjil61h2sw78h6b4lqlgf867r1bsqkml-nss-3.68.3/lib:/nix/store/vz2d55s4qclrds6lvr0n86dlaxw0hha2-pango-1.50.6/lib:/nix/store/v8yddpay5ilm3m59yj64pdnc13gjkav1-pipewire-0.3.49-lib/lib:/nix/store/k9pdm7kwgsw6cbpzcaibr5yszldg74vc-systemd-250.4/lib:/nix/store/10532vjygwv7k9yvrwy1ncrkr5aw4jbd-wayland-1.20.0/lib:/nix/store/7h56w8gbvdcx1gmhn1x4rkb1952bds2s-xdg-utils-unstable-2020-10-21/lib:/nix/store/ksgadr23jjc6i0qy6nlp3xmhvpzdlqys-libxcb-1.14/lib:/nix/store/vybkpg9gqy10d8pm17wz6x1ljrcsfn5d-zlib-1.2.12/lib:/nix/store/5ydza30xl0c77my0rhpbf8j0mbp75fsm-snappy-1.1.9/lib:/nix/store/qs6p4d1pd28km8906rxw22dvssfs49m4-libpulseaudio-15.0/lib:/nix/store/vs1qkjqz3rymj8mybpibgdr8c3jppwj9-libva-2.14.0/lib:/nix/store/a3nqf29wdxf2022ri3m3y8302mvlpfy7-alsa-lib-1.2.6.1/lib64:/nix/store/60wrl1spfjm04hsqym5jqanc0sgz3qs3-at-spi2-atk-2.38.0/lib64:/nix/store/5s9n8syy1fa22jxmg865xpivb63kp1hy-at-spi2-core-2.44.1/lib64:/nix/store/v8j970izrm2agqqh19c6j23d25asip3d-atk-2.36.0/lib64:/nix/store/fnad9qhmlzl9bgm2wh6mgv0fmrg33k32-cairo-1.16.0/lib64:/nix/store/byzr6wj4dsh6y146jxxwrhc64lq92bmx-cups-2.4.1-lib/lib64:/nix/store/hyqhzp899sr3zh0z1fsy8wl59mj2ib1l-dbus-1.12.20-lib/lib64:/nix/store/yabw7v3z07l2n7jnhma0mjkr7h586a7b-expat-2.4.7/lib64:/nix/store/4bhkj5w5319pig0n47l2qhalx2bz9f0c-fontconfig-2.13.94-lib/lib64:/nix/store/pjfijj9j1pwcb76yqny64k8wk7yz191p-freetype-2.12.0/lib64:/nix/store/nyxva3vqcx7331i61760h4sdkmxnrkzh-gdk-pixbuf-2.42.8/lib64:/nix/store/hqvf17nn5fmwrlrgq9qyplz1id923jri-glib-2.72.1/lib64:/nix/store/3a3j7xarxmrarxasi2sr05afx3vnx6mr-gtk+3-3.24.33/lib64:/nix/store/5fl91pknshzhgv1147j2bfyr2jr77skq-libdrm-2.4.110/lib64:/nix/store/3lrppnbkc5wlnh5ifwlxhghb0lxan42h-libX11-1.7.2/lib64:/nix/store/xsi5z55nxm3n3m0pzysq8c6ccrqqhj4z-libGL-1.4.0/lib64:/nix/store/8g1dgdpjgldsa4faj7rfnw06n04idhyz-libxkbcommon-1.4.0/lib64:/nix/store/gwd0iijizj8k4ba5wpxnwjjgn1sfc1sp-libXScrnSaver-1.2.3/lib64:/nix/store/zgjj5alj0qyhmgy45rfdbd35zgqw1syz-libXcomposite-0.4.5/lib64:/nix/store/ipgwk0ywkbrwsfqi6s0nmbybqds84zkd-libXcursor-1.2.0/lib64:/nix/store/fxah0qfxr07y3j2sldq2izfq3h5j1wiy-libXdamage-1.1.5/lib64:/nix/store/91x9nnlbqpqg8mcb9qjlb1q9jkq9v7v3-libXext-1.3.4/lib64:/nix/store/llnvb4l7krvq2s0rsbxb1r2xn3s8hz23-libXfixes-6.0.0/lib64:/nix/store/nw60c3l2qq3akr1ld6z0rlzb9gw293h8-libXi-1.8/lib64:/nix/store/85c71b4h8afbhp648mbm2lnq73ipmsc2-libXrandr-1.5.2/lib64:/nix/store/89yf477csc3hyn7kpvc1nnj1731n2sml-libXrender-0.9.10/lib64:/nix/store/h05nwiqmiz17484qcj30m4xjsm8yf5lx-libxshmfence-1.3/lib64:/nix/store/2wa7lfl6kh0q9p3njxdjzlfkvv8lhmn0-libXtst-1.2.3/lib64:/nix/store/ip19s520xvvxhpm69g355vbhgyvv4z7q-util-linux-minimal-2.37.4-lib/lib64:/nix/store/cyxpd4cf7wbb5q18bcqkkyhriyajas11-mesa-22.0.2/lib64:/nix/store/l6np4r402hwcj0kcqgrv5mqc819qnbah-nspr-4.33/lib64:/nix/store/jjil61h2sw78h6b4lqlgf867r1bsqkml-nss-3.68.3/lib64:/nix/store/vz2d55s4qclrds6lvr0n86dlaxw0hha2-pango-1.50.6/lib64:/nix/store/v8yddpay5ilm3m59yj64pdnc13gjkav1-pipewire-0.3.49-lib/lib64:/nix/store/k9pdm7kwgsw6cbpzcaibr5yszldg74vc-systemd-250.4/lib64:/nix/store/10532vjygwv7k9yvrwy1ncrkr5aw4jbd-wayland-1.20.0/lib64:/nix/store/7h56w8gbvdcx1gmhn1x4rkb1952bds2s-xdg-utils-unstable-2020-10-21/lib64:/nix/store/ksgadr23jjc6i0qy6nlp3xmhvpzdlqys-libxcb-1.14/lib64:/nix/store/vybkpg9gqy10d8pm17wz6x1ljrcsfn5d-zlib-1.2.12/lib64:/nix/store/5ydza30xl0c77my0rhpbf8j0mbp75fsm-snappy-1.1.9/lib64:/nix/store/qs6p4d1pd28km8906rxw22dvssfs49m4-libpulseaudio-15.0/lib64:/nix/store/vs1qkjqz3rymj8mybpibgdr8c3jppwj9-libva-2.14.0/lib64");
    set_env_prefix("PATH", ":", "/nix/store/a3nqf29wdxf2022ri3m3y8302mvlpfy7-alsa-lib-1.2.6.1/bin:/nix/store/60wrl1spfjm04hsqym5jqanc0sgz3qs3-at-spi2-atk-2.38.0/bin:/nix/store/5s9n8syy1fa22jxmg865xpivb63kp1hy-at-spi2-core-2.44.1/bin:/nix/store/v8j970izrm2agqqh19c6j23d25asip3d-atk-2.36.0/bin:/nix/store/fnad9qhmlzl9bgm2wh6mgv0fmrg33k32-cairo-1.16.0/bin:/nix/store/4xjyn0y5vjwa6dddbwg9ld3ybzrfmykz-cups-2.4.1/bin:/nix/store/ijwf72dh14h3c805cfil16vq3s1lnv4g-dbus-1.12.20/bin:/nix/store/yabw7v3z07l2n7jnhma0mjkr7h586a7b-expat-2.4.7/bin:/nix/store/hz5ykkzkspx0zcvfk1v18024rbmg1pgq-fontconfig-2.13.94-bin/bin:/nix/store/pjfijj9j1pwcb76yqny64k8wk7yz191p-freetype-2.12.0/bin:/nix/store/nyxva3vqcx7331i61760h4sdkmxnrkzh-gdk-pixbuf-2.42.8/bin:/nix/store/jj5pjm2dgwi8w9rgnaz75mnbkds0pazx-glib-2.72.1-bin/bin:/nix/store/3a3j7xarxmrarxasi2sr05afx3vnx6mr-gtk+3-3.24.33/bin:/nix/store/jnw5h21iq9ambh1z0ky4dlxz8jmp4jss-libdrm-2.4.110-bin/bin:/nix/store/3lrppnbkc5wlnh5ifwlxhghb0lxan42h-libX11-1.7.2/bin:/nix/store/xsi5z55nxm3n3m0pzysq8c6ccrqqhj4z-libGL-1.4.0/bin:/nix/store/8g1dgdpjgldsa4faj7rfnw06n04idhyz-libxkbcommon-1.4.0/bin:/nix/store/gwd0iijizj8k4ba5wpxnwjjgn1sfc1sp-libXScrnSaver-1.2.3/bin:/nix/store/zgjj5alj0qyhmgy45rfdbd35zgqw1syz-libXcomposite-0.4.5/bin:/nix/store/ipgwk0ywkbrwsfqi6s0nmbybqds84zkd-libXcursor-1.2.0/bin:/nix/store/fxah0qfxr07y3j2sldq2izfq3h5j1wiy-libXdamage-1.1.5/bin:/nix/store/91x9nnlbqpqg8mcb9qjlb1q9jkq9v7v3-libXext-1.3.4/bin:/nix/store/llnvb4l7krvq2s0rsbxb1r2xn3s8hz23-libXfixes-6.0.0/bin:/nix/store/nw60c3l2qq3akr1ld6z0rlzb9gw293h8-libXi-1.8/bin:/nix/store/85c71b4h8afbhp648mbm2lnq73ipmsc2-libXrandr-1.5.2/bin:/nix/store/89yf477csc3hyn7kpvc1nnj1731n2sml-libXrender-0.9.10/bin:/nix/store/h05nwiqmiz17484qcj30m4xjsm8yf5lx-libxshmfence-1.3/bin:/nix/store/2wa7lfl6kh0q9p3njxdjzlfkvv8lhmn0-libXtst-1.2.3/bin:/nix/store/hx6rbk17wcf70q439g0lv6v18vfhq1cc-util-linux-minimal-2.37.4-bin/bin:/nix/store/cyxpd4cf7wbb5q18bcqkkyhriyajas11-mesa-22.0.2/bin:/nix/store/l6np4r402hwcj0kcqgrv5mqc819qnbah-nspr-4.33/bin:/nix/store/jjil61h2sw78h6b4lqlgf867r1bsqkml-nss-3.68.3/bin:/nix/store/8y5sjwfks4m3irbrbp4rfjm3nay7xcm6-pango-1.50.6-bin/bin:/nix/store/pmyn5fb46npgwxa3hdhsb284yflxh6lh-pipewire-0.3.49/bin:/nix/store/k9pdm7kwgsw6cbpzcaibr5yszldg74vc-systemd-250.4/bin:/nix/store/hsm8f4a9vh02k91djc5dlr3ljwv060rl-wayland-1.20.0-bin/bin:/nix/store/7h56w8gbvdcx1gmhn1x4rkb1952bds2s-xdg-utils-unstable-2020-10-21/bin:/nix/store/ksgadr23jjc6i0qy6nlp3xmhvpzdlqys-libxcb-1.14/bin:/nix/store/vybkpg9gqy10d8pm17wz6x1ljrcsfn5d-zlib-1.2.12/bin:/nix/store/5ydza30xl0c77my0rhpbf8j0mbp75fsm-snappy-1.1.9/bin:/nix/store/qs6p4d1pd28km8906rxw22dvssfs49m4-libpulseaudio-15.0/bin:/nix/store/vs1qkjqz3rymj8mybpibgdr8c3jppwj9-libva-2.14.0/bin");
    char **argv_tmp = calloc(4 + argc, sizeof(*argv_tmp));
    assert(argv_tmp != NULL);
    argv_tmp[0] = argv[0];
    argv_tmp[1] = "--enable-features=VaapiVideoDecoder,VaapiVideoEncoder";
    argv_tmp[2] = "--disable-features=UseChromeOSDirectVideoDecoder";
    argv_tmp[3] = "${NIXOS_OZONE_WL:+${WAYLAND_DISPLAY:+--ozone-platform=wayland}}";
    for (int i = 1; i < argc; ++i) {
        argv_tmp[3 + i] = argv[i];
    }
    argv_tmp[3 + argc] = NULL;
    argv = argv_tmp;
    return execv("/nix/store/vmyri2fwck810hsfm7cf4jaa2hv8mnzj-brave-1.38.109/bin/.brave-wrapped", argv);
}
const char * DOCSTRING = "\n\n\n# ------------------------------------------------------------------------------------\n# The C-code for this binary wrapper has been generated using the following command:\n\n\nmakeCWrapper /nix/store/vmyri2fwck810hsfm7cf4jaa2hv8mnzj-brave-1.38.109/bin/.brave-wrapped \\\n    --inherit-argv0 \\\n    --prefix 'GIO_EXTRA_MODULES' ':' '/nix/store/q1yars9wd0qyx1ib78l4y6jc3cfdd5l8-dconf-0.40.0-lib/lib/gio/modules' \\\n    --prefix 'GIO_EXTRA_MODULES' ':' '/nix/store/q1yars9wd0qyx1ib78l4y6jc3cfdd5l8-dconf-0.40.0-lib/lib/gio/modules' \\\n    --set 'GDK_PIXBUF_MODULE_FILE' '/nix/store/k6a58j0i1qy9vc87lppz1gwq29drr62b-librsvg-2.54.0/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache' \\\n    --prefix 'XDG_DATA_DIRS' ':' '/nix/store/3a3j7xarxmrarxasi2sr05afx3vnx6mr-gtk+3-3.24.33/share/gsettings-schemas/gtk+3-3.24.33:/nix/store/snpvaw1f86407x7aw114khhpvmyjcr0g-gsettings-desktop-schemas-42.0/share/gsettings-schemas/gsettings-desktop-schemas-42.0' \\\n    --prefix 'XDG_DATA_DIRS' ':' '/nix/store/vmyri2fwck810hsfm7cf4jaa2hv8mnzj-brave-1.38.109/share' \\\n    --prefix 'LD_LIBRARY_PATH' ':' '/nix/store/a3nqf29wdxf2022ri3m3y8302mvlpfy7-alsa-lib-1.2.6.1/lib:/nix/store/60wrl1spfjm04hsqym5jqanc0sgz3qs3-at-spi2-atk-2.38.0/lib:/nix/store/5s9n8syy1fa22jxmg865xpivb63kp1hy-at-spi2-core-2.44.1/lib:/nix/store/v8j970izrm2agqqh19c6j23d25asip3d-atk-2.36.0/lib:/nix/store/fnad9qhmlzl9bgm2wh6mgv0fmrg33k32-cairo-1.16.0/lib:/nix/store/byzr6wj4dsh6y146jxxwrhc64lq92bmx-cups-2.4.1-lib/lib:/nix/store/hyqhzp899sr3zh0z1fsy8wl59mj2ib1l-dbus-1.12.20-lib/lib:/nix/store/yabw7v3z07l2n7jnhma0mjkr7h586a7b-expat-2.4.7/lib:/nix/store/4bhkj5w5319pig0n47l2qhalx2bz9f0c-fontconfig-2.13.94-lib/lib:/nix/store/pjfijj9j1pwcb76yqny64k8wk7yz191p-freetype-2.12.0/lib:/nix/store/nyxva3vqcx7331i61760h4sdkmxnrkzh-gdk-pixbuf-2.42.8/lib:/nix/store/hqvf17nn5fmwrlrgq9qyplz1id923jri-glib-2.72.1/lib:/nix/store/3a3j7xarxmrarxasi2sr05afx3vnx6mr-gtk+3-3.24.33/lib:/nix/store/5fl91pknshzhgv1147j2bfyr2jr77skq-libdrm-2.4.110/lib:/nix/store/3lrppnbkc5wlnh5ifwlxhghb0lxan42h-libX11-1.7.2/lib:/nix/store/xsi5z55nxm3n3m0pzysq8c6ccrqqhj4z-libGL-1.4.0/lib:/nix/store/8g1dgdpjgldsa4faj7rfnw06n04idhyz-libxkbcommon-1.4.0/lib:/nix/store/gwd0iijizj8k4ba5wpxnwjjgn1sfc1sp-libXScrnSaver-1.2.3/lib:/nix/store/zgjj5alj0qyhmgy45rfdbd35zgqw1syz-libXcomposite-0.4.5/lib:/nix/store/ipgwk0ywkbrwsfqi6s0nmbybqds84zkd-libXcursor-1.2.0/lib:/nix/store/fxah0qfxr07y3j2sldq2izfq3h5j1wiy-libXdamage-1.1.5/lib:/nix/store/91x9nnlbqpqg8mcb9qjlb1q9jkq9v7v3-libXext-1.3.4/lib:/nix/store/llnvb4l7krvq2s0rsbxb1r2xn3s8hz23-libXfixes-6.0.0/lib:/nix/store/nw60c3l2qq3akr1ld6z0rlzb9gw293h8-libXi-1.8/lib:/nix/store/85c71b4h8afbhp648mbm2lnq73ipmsc2-libXrandr-1.5.2/lib:/nix/store/89yf477csc3hyn7kpvc1nnj1731n2sml-libXrender-0.9.10/lib:/nix/store/h05nwiqmiz17484qcj30m4xjsm8yf5lx-libxshmfence-1.3/lib:/nix/store/2wa7lfl6kh0q9p3njxdjzlfkvv8lhmn0-libXtst-1.2.3/lib:/nix/store/ip19s520xvvxhpm69g355vbhgyvv4z7q-util-linux-minimal-2.37.4-lib/lib:/nix/store/cyxpd4cf7wbb5q18bcqkkyhriyajas11-mesa-22.0.2/lib:/nix/store/l6np4r402hwcj0kcqgrv5mqc819qnbah-nspr-4.33/lib:/nix/store/jjil61h2sw78h6b4lqlgf867r1bsqkml-nss-3.68.3/lib:/nix/store/vz2d55s4qclrds6lvr0n86dlaxw0hha2-pango-1.50.6/lib:/nix/store/v8yddpay5ilm3m59yj64pdnc13gjkav1-pipewire-0.3.49-lib/lib:/nix/store/k9pdm7kwgsw6cbpzcaibr5yszldg74vc-systemd-250.4/lib:/nix/store/10532vjygwv7k9yvrwy1ncrkr5aw4jbd-wayland-1.20.0/lib:/nix/store/7h56w8gbvdcx1gmhn1x4rkb1952bds2s-xdg-utils-unstable-2020-10-21/lib:/nix/store/ksgadr23jjc6i0qy6nlp3xmhvpzdlqys-libxcb-1.14/lib:/nix/store/vybkpg9gqy10d8pm17wz6x1ljrcsfn5d-zlib-1.2.12/lib:/nix/store/5ydza30xl0c77my0rhpbf8j0mbp75fsm-snappy-1.1.9/lib:/nix/store/qs6p4d1pd28km8906rxw22dvssfs49m4-libpulseaudio-15.0/lib:/nix/store/vs1qkjqz3rymj8mybpibgdr8c3jppwj9-libva-2.14.0/lib:/nix/store/a3nqf29wdxf2022ri3m3y8302mvlpfy7-alsa-lib-1.2.6.1/lib64:/nix/store/60wrl1spfjm04hsqym5jqanc0sgz3qs3-at-spi2-atk-2.38.0/lib64:/nix/store/5s9n8syy1fa22jxmg865xpivb63kp1hy-at-spi2-core-2.44.1/lib64:/nix/store/v8j970izrm2agqqh19c6j23d25asip3d-atk-2.36.0/lib64:/nix/store/fnad9qhmlzl9bgm2wh6mgv0fmrg33k32-cairo-1.16.0/lib64:/nix/store/byzr6wj4dsh6y146jxxwrhc64lq92bmx-cups-2.4.1-lib/lib64:/nix/store/hyqhzp899sr3zh0z1fsy8wl59mj2ib1l-dbus-1.12.20-lib/lib64:/nix/store/yabw7v3z07l2n7jnhma0mjkr7h586a7b-expat-2.4.7/lib64:/nix/store/4bhkj5w5319pig0n47l2qhalx2bz9f0c-fontconfig-2.13.94-lib/lib64:/nix/store/pjfijj9j1pwcb76yqny64k8wk7yz191p-freetype-2.12.0/lib64:/nix/store/nyxva3vqcx7331i61760h4sdkmxnrkzh-gdk-pixbuf-2.42.8/lib64:/nix/store/hqvf17nn5fmwrlrgq9qyplz1id923jri-glib-2.72.1/lib64:/nix/store/3a3j7xarxmrarxasi2sr05afx3vnx6mr-gtk+3-3.24.33/lib64:/nix/store/5fl91pknshzhgv1147j2bfyr2jr77skq-libdrm-2.4.110/lib64:/nix/store/3lrppnbkc5wlnh5ifwlxhghb0lxan42h-libX11-1.7.2/lib64:/nix/store/xsi5z55nxm3n3m0pzysq8c6ccrqqhj4z-libGL-1.4.0/lib64:/nix/store/8g1dgdpjgldsa4faj7rfnw06n04idhyz-libxkbcommon-1.4.0/lib64:/nix/store/gwd0iijizj8k4ba5wpxnwjjgn1sfc1sp-libXScrnSaver-1.2.3/lib64:/nix/store/zgjj5alj0qyhmgy45rfdbd35zgqw1syz-libXcomposite-0.4.5/lib64:/nix/store/ipgwk0ywkbrwsfqi6s0nmbybqds84zkd-libXcursor-1.2.0/lib64:/nix/store/fxah0qfxr07y3j2sldq2izfq3h5j1wiy-libXdamage-1.1.5/lib64:/nix/store/91x9nnlbqpqg8mcb9qjlb1q9jkq9v7v3-libXext-1.3.4/lib64:/nix/store/llnvb4l7krvq2s0rsbxb1r2xn3s8hz23-libXfixes-6.0.0/lib64:/nix/store/nw60c3l2qq3akr1ld6z0rlzb9gw293h8-libXi-1.8/lib64:/nix/store/85c71b4h8afbhp648mbm2lnq73ipmsc2-libXrandr-1.5.2/lib64:/nix/store/89yf477csc3hyn7kpvc1nnj1731n2sml-libXrender-0.9.10/lib64:/nix/store/h05nwiqmiz17484qcj30m4xjsm8yf5lx-libxshmfence-1.3/lib64:/nix/store/2wa7lfl6kh0q9p3njxdjzlfkvv8lhmn0-libXtst-1.2.3/lib64:/nix/store/ip19s520xvvxhpm69g355vbhgyvv4z7q-util-linux-minimal-2.37.4-lib/lib64:/nix/store/cyxpd4cf7wbb5q18bcqkkyhriyajas11-mesa-22.0.2/lib64:/nix/store/l6np4r402hwcj0kcqgrv5mqc819qnbah-nspr-4.33/lib64:/nix/store/jjil61h2sw78h6b4lqlgf867r1bsqkml-nss-3.68.3/lib64:/nix/store/vz2d55s4qclrds6lvr0n86dlaxw0hha2-pango-1.50.6/lib64:/nix/store/v8yddpay5ilm3m59yj64pdnc13gjkav1-pipewire-0.3.49-lib/lib64:/nix/store/k9pdm7kwgsw6cbpzcaibr5yszldg74vc-systemd-250.4/lib64:/nix/store/10532vjygwv7k9yvrwy1ncrkr5aw4jbd-wayland-1.20.0/lib64:/nix/store/7h56w8gbvdcx1gmhn1x4rkb1952bds2s-xdg-utils-unstable-2020-10-21/lib64:/nix/store/ksgadr23jjc6i0qy6nlp3xmhvpzdlqys-libxcb-1.14/lib64:/nix/store/vybkpg9gqy10d8pm17wz6x1ljrcsfn5d-zlib-1.2.12/lib64:/nix/store/5ydza30xl0c77my0rhpbf8j0mbp75fsm-snappy-1.1.9/lib64:/nix/store/qs6p4d1pd28km8906rxw22dvssfs49m4-libpulseaudio-15.0/lib64:/nix/store/vs1qkjqz3rymj8mybpibgdr8c3jppwj9-libva-2.14.0/lib64' \\\n    --prefix 'PATH' ':' '/nix/store/a3nqf29wdxf2022ri3m3y8302mvlpfy7-alsa-lib-1.2.6.1/bin:/nix/store/60wrl1spfjm04hsqym5jqanc0sgz3qs3-at-spi2-atk-2.38.0/bin:/nix/store/5s9n8syy1fa22jxmg865xpivb63kp1hy-at-spi2-core-2.44.1/bin:/nix/store/v8j970izrm2agqqh19c6j23d25asip3d-atk-2.36.0/bin:/nix/store/fnad9qhmlzl9bgm2wh6mgv0fmrg33k32-cairo-1.16.0/bin:/nix/store/4xjyn0y5vjwa6dddbwg9ld3ybzrfmykz-cups-2.4.1/bin:/nix/store/ijwf72dh14h3c805cfil16vq3s1lnv4g-dbus-1.12.20/bin:/nix/store/yabw7v3z07l2n7jnhma0mjkr7h586a7b-expat-2.4.7/bin:/nix/store/hz5ykkzkspx0zcvfk1v18024rbmg1pgq-fontconfig-2.13.94-bin/bin:/nix/store/pjfijj9j1pwcb76yqny64k8wk7yz191p-freetype-2.12.0/bin:/nix/store/nyxva3vqcx7331i61760h4sdkmxnrkzh-gdk-pixbuf-2.42.8/bin:/nix/store/jj5pjm2dgwi8w9rgnaz75mnbkds0pazx-glib-2.72.1-bin/bin:/nix/store/3a3j7xarxmrarxasi2sr05afx3vnx6mr-gtk+3-3.24.33/bin:/nix/store/jnw5h21iq9ambh1z0ky4dlxz8jmp4jss-libdrm-2.4.110-bin/bin:/nix/store/3lrppnbkc5wlnh5ifwlxhghb0lxan42h-libX11-1.7.2/bin:/nix/store/xsi5z55nxm3n3m0pzysq8c6ccrqqhj4z-libGL-1.4.0/bin:/nix/store/8g1dgdpjgldsa4faj7rfnw06n04idhyz-libxkbcommon-1.4.0/bin:/nix/store/gwd0iijizj8k4ba5wpxnwjjgn1sfc1sp-libXScrnSaver-1.2.3/bin:/nix/store/zgjj5alj0qyhmgy45rfdbd35zgqw1syz-libXcomposite-0.4.5/bin:/nix/store/ipgwk0ywkbrwsfqi6s0nmbybqds84zkd-libXcursor-1.2.0/bin:/nix/store/fxah0qfxr07y3j2sldq2izfq3h5j1wiy-libXdamage-1.1.5/bin:/nix/store/91x9nnlbqpqg8mcb9qjlb1q9jkq9v7v3-libXext-1.3.4/bin:/nix/store/llnvb4l7krvq2s0rsbxb1r2xn3s8hz23-libXfixes-6.0.0/bin:/nix/store/nw60c3l2qq3akr1ld6z0rlzb9gw293h8-libXi-1.8/bin:/nix/store/85c71b4h8afbhp648mbm2lnq73ipmsc2-libXrandr-1.5.2/bin:/nix/store/89yf477csc3hyn7kpvc1nnj1731n2sml-libXrender-0.9.10/bin:/nix/store/h05nwiqmiz17484qcj30m4xjsm8yf5lx-libxshmfence-1.3/bin:/nix/store/2wa7lfl6kh0q9p3njxdjzlfkvv8lhmn0-libXtst-1.2.3/bin:/nix/store/hx6rbk17wcf70q439g0lv6v18vfhq1cc-util-linux-minimal-2.37.4-bin/bin:/nix/store/cyxpd4cf7wbb5q18bcqkkyhriyajas11-mesa-22.0.2/bin:/nix/store/l6np4r402hwcj0kcqgrv5mqc819qnbah-nspr-4.33/bin:/nix/store/jjil61h2sw78h6b4lqlgf867r1bsqkml-nss-3.68.3/bin:/nix/store/8y5sjwfks4m3irbrbp4rfjm3nay7xcm6-pango-1.50.6-bin/bin:/nix/store/pmyn5fb46npgwxa3hdhsb284yflxh6lh-pipewire-0.3.49/bin:/nix/store/k9pdm7kwgsw6cbpzcaibr5yszldg74vc-systemd-250.4/bin:/nix/store/hsm8f4a9vh02k91djc5dlr3ljwv060rl-wayland-1.20.0-bin/bin:/nix/store/7h56w8gbvdcx1gmhn1x4rkb1952bds2s-xdg-utils-unstable-2020-10-21/bin:/nix/store/ksgadr23jjc6i0qy6nlp3xmhvpzdlqys-libxcb-1.14/bin:/nix/store/vybkpg9gqy10d8pm17wz6x1ljrcsfn5d-zlib-1.2.12/bin:/nix/store/5ydza30xl0c77my0rhpbf8j0mbp75fsm-snappy-1.1.9/bin:/nix/store/qs6p4d1pd28km8906rxw22dvssfs49m4-libpulseaudio-15.0/bin:/nix/store/vs1qkjqz3rymj8mybpibgdr8c3jppwj9-libva-2.14.0/bin' \\\n    --add-flags '--enable-features=VaapiVideoDecoder,VaapiVideoEncoder' \\\n    --add-flags '--disable-features=UseChromeOSDirectVideoDecoder' \\\n    --add-flags '${NIXOS_OZONE_WL:+${WAYLAND_DISPLAY:+--ozone-platform=wayland}}'\n\n\n# (Use `nix-shell -p makeBinaryWrapper` to get access to makeCWrapper in your shell)\n# ------------------------------------------------------------------------------------\n\n\n";
```
</details>

~(Should fixes like these go to master or staging-next? On one hand it's needed ASAP in staging-next, but on the other hand it might fix issues in master, and there's only a 6h delay between the two...)~

ncfavier committed:
makeBinaryWrapper: add -Wno-overlength-strings

The generated C code contains large string literals that are longer than
the maximum length specified by the standard.

However, GCC has no trouble dealing with those strings, so we can just
add -Wno-overlength-strings.

https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Woverlength-strings

vcunat commented:
So maybe #171726 was just this and should be reverted?

ncfavier commented:
I don't know, was https://github.com/NixOS/nixpkgs/pull/171726 confirmed to fix an issue? If it did, it should probably be left alone since `NIX_CFLAGS_COMPILE` is not [supposed](https://github.com/NixOS/nixpkgs/pull/164163/commits/f8cc8ff5755528d9a73671481ba3d6fb00f4e8d5) to interfere with the wrapper.

bergkvist commented:
Does this also work for clang on macOS?

I guess it might be useful to add a test specifically with really long strings.

That way, this problem will be immediately evident if we try to make binary wrappers work for another OS/compiler in the future - or if a gcc/clang changes its behavior in the future.

vcunat commented:
Yes, that PR fixed builds: https://hydra.nixos.org/eval/1760132?filter=gnome.&compare=1760093&full=#tabs-now-succeed

vcunat commented:
Total rebuild amount (by OfBorg) is over 27k.  Certainly not for direct merge to `master`.  At this stage I'm even reluctant about `staging-next`, but probably... due to the regressions.

ncfavier commented:
OK. Indeed that PR seems to address the wrapper issues, I'll have to investigate why it works.

I'll add a test.

vcunat commented:
You're using `stdenv.cc`, i.e. wrapped compiler, so that's why it reacts to stuff like `$NIX_CFLAGS_COMPILE`.

ncfavier commented:
Yes, that was the whole point of https://github.com/NixOS/nixpkgs/pull/164163/commits/f8cc8ff5755528d9a73671481ba3d6fb00f4e8d5. But apparently Bash does not work as expected:

```bash
export FOO=bar
f() { local FOO; unset FOO; declare -p FOO; env | grep FOO; }
f
```

yields

```
declare -- FOO
FOO=bar
```

I'm not sure what's happening since that commit definitely fixed the issue for libreoffice.

fabianhjr commented:
> So maybe https://github.com/NixOS/nixpkgs/pull/171726 was just this and should be reverted?

I think so, yeah it should be reverted if this is merged and this is a more general fix.

vcunat commented:
Maybe the `local` part makes the change invisible in child processes?

ncfavier commented:
Well, `f() { local NIX_CFLAGS_COMPILE=; ... }` has the desired behaviour, so it seems more like an `unset` quirk. Empty or unset doesn't matter here, so that will do.

ncfavier committed:
makeBinaryWrapper: add overlength-strings test

ncfavier committed:
makeBinaryWrapper: really unset NIX_CFLAGS

https://github.com/NixOS/nixpkgs/commit/f8cc8ff5755528d9a73671481ba3d6fb00f4e8d5
fails to unset the variables in the environment due to a Bash quirk,
so set them to the empty string instead.

ncfavier committed:
Revert "gnome: fix compilation with gcc 11.3.0"

This reverts commit 251ab5f3f61e2402833291d28ca6a19ec8839e2a.

Fixed by https://github.com/NixOS/nixpkgs/pull/171819 instead.

ncfavier commented on pkgs/test/make-binary-wrapper/overlength-strings.cmdline:
```diff
@@ -0,0 +1 @@
+    --prefix PATH : /nix/store/00000000000000000000000000000000-loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo ...
```
I think it would complicate the test infrastructure for very little value added

vcunat replied:
It might also be over `PATH_MAX` (on some platforms; restricts a single path to a file/dir), but that probably doesn't matter here.

ncfavier replied:
Right, we're only looking at the value in the environment, not using it.
